api/test/e2e/journeys/test_schemas.py

```diff
@@ -52,23 +45,116 @@ def update_schema(self, schema: dict) -> Response:
             json=schema,
         )
 
+    def get_dataset_from_dynamodb(self, layer: str, domain: str, dataset: str) -> dict:
+        dynamodb = boto3.resource("dynamodb", region_name=AWS_REGION)
+        table = dynamodb.Table(SCHEMA_TABLE_NAME)
+        try:
+            response = table.query(
+                KeyConditionExpression=Key("PK").eq(f"{layer}/{domain}/{dataset}"),
+            )
+        except ClientError as e:
+            raise e
+        return response["Items"]
+
+    def extract_column_names(self, dataset_schema: dict) -> list:
+        return [item["name"] for item in dataset_schema[0]["columns"]]
```

Comment:
Why are we selecting the first element of dataset_schema? The type says it's a dict